### PR TITLE
Fix event monitor panic.

### PR DESCRIPTION
**Pull request by Random-Liu**
Fix the event monitor panic.

We should only stop ticker when it is not nil.

Since `start` and `stop` can be called at the same time, we need a lock to protect it.

Signed-off-by: Lantao Liu <lantaol@google.com>

### CODE REVIEW

**Comment by dmcgowan** [comment]
LGTM

**Comment by stevvooe** [comment]
LGTM

File: pkg/server/events.go [hunk 1]
```diff
@@ -56,7 +57,9 @@ type eventMonitor struct {
 }
 
 type backOff struct {
-	queuePool     map[string]*backOffQueue
+	queuePool map[string]*backOffQueue
```
Comment (yanxuean) [inline pkg/server/events.go]: go fmt

Comment (Random-Liu) [reply pkg/server/events.go]: That is `gofmt` result.

File: pkg/server/events.go [hunk 2]
```diff
@@ -366,12 +369,18 @@ func (b *backOff) reBackOff(key string, events []interface{}, oldDuration time.D
...
 	return b.ticker.C
 }
 
 func (b *backOff) stop() {
-	b.ticker.Stop()
+	b.tickerMu.Lock()
+	defer b.tickerMu.Unlock()
+	if b.ticker != nil {
+		b.ticker.Stop()
+	}
```
Comment (mikebrow) [inline pkg/server/events.go]: else...logrus.WithError(err).Error("attempt to stop ticker before start has completed allocating the ticker")

note: should be impossible for this to happen now that you have a mutex... if the log is impossible we don't need the nil check..

File: pkg/server/events.go [hunk 3]
```diff
@@ -120,8 +123,8 @@ func (em *eventMonitor) start() (<-chan struct{}, error) {
 		return nil, errors.New("event channel is nil")
 	}
 	closeCh := make(chan struct{})
+	backOffCheckCh := em.backOff.start()
```
Comment (mikebrow) [inline pkg/server/events.go]: good!

File: pkg/server/events.go (hunk 2 above)
Comment (mikebrow) [reply pkg/server/events.go]: user interrupts during init of the plugins... interesting thx

**Pull request merged**